src/server/routes/members/index.ts

```diff
@@ -105,6 +117,28 @@ async function createCoconutVideoEncodingJob(
   );
 }
 
+async function moveInviteVideoToPublicVideo(
+  config: Config,
+  storage: BucketStorage,
+  memberUid: string,
+  videoToken?: string
+) {
+  const publicVideoRef = getPublicVideoRef(config, storage, memberUid);
+  if ((await publicVideoRef.exists())[0]) {
+    throw new HttpApiError(
+      httpStatus.BAD_REQUEST,
+      "Video already exists at intended storage destination. Cannot overwrite.",
+      {}
+    );
+  }
+
+  const inviteVideoRef = videoToken
+    ? getInviteVideoRef(config, storage, videoToken)
+    : getPrivateVideoRef(config, storage, memberUid);
```

Comment:
In what scenario will the videoToken not be passed to this endpoint? If the privateVideo is still going to have metadata attached, then we can't just move it into the publicVideo bucket, right?

Reply:
Oh --- is this the case of whether the inviter/invitee recorded the video?

Reply:
If a new user downloaded the app and went through the non-deeplink onboarding flow where they record their video from their own log in, it would be in private-video. This wouldn't have the videoToken. It also shouldn't have metadata attached since the compression applies to both flows.

If the video was recorded from the inviter's side, videoToken will be passed in and the video would be in invite-video.

Reply:
gotcha, cool cool